### PR TITLE
Pandas deprecation fixes

### PR DESCRIPTION
**Pull request by koen-vg**
Just fixing some deprecation and future warnings as I go along :)

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [ ] Changed dependencies are added to `envs/environment.yaml`.
- [ ] Changes in configuration options are added in all of `config.default.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [ ] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

**Comment by koen-vg** [comment]
Ah, this time the tests saved me; I was a little too eager with just one of the fixes.

File: scripts/make_summary_perfect.py [hunk 1]
```diff
@@ -514,9 +508,7 @@ def calculate_weighted_prices(n, label, weighted_prices):
             if names.empty:
                 continue
 
-            load += (
-                n.links_t.p0[names].groupby(n.links.loc[names, "bus0"], axis=1).sum()
-            )
+            load += n.links_t.p0[names].T.groupby(n.links.loc[names, "bus0"]).sum()
```
Comment (fneum) [inline scripts/make_summary_perfect.py:511]: Just double-checking the transposes here:

So far, in most cases `.groupby(..., axis=1).sum()` is replaced by `.T.groupby(...).sum().T`

File: scripts/make_summary_perfect.py [hunk 2]
```diff
@@ -652,7 +644,7 @@ def calculate_co2_emissions(n, label, df):
         emitted = n.generators_t.p[gens.index].mul(em_pu)
 
         emitted_grouped = (
-            emitted.groupby(level=0).sum().groupby(n.generators.carrier, axis=1).sum().T
+            emitted.groupby(level=0).sum().T.groupby(n.generators.carrier).sum().T
```
Comment (fneum) [inline scripts/make_summary_perfect.py:647]: similar here, given the additional transpose

**Review by fneum — COMMENTED** [review]
Looks good! Thanks! Just unsure about two cases that I couldn't verify without testing it myself.

**Comment by koen-vg** [comment]
Yes you were right! At least the one I was able to test, in the weighted price calculation. It turned out the code path wasn't used before due to an unrelated bug, which I now also fixed. I will say that the whole function seems pretty fragile...

For the CO2 emissions calculations, the code path you pointed out is also never taken! Hence why I didn't find the mistake. It turns out that in sector-coupled perfect foresight networks there aren't any co2-emitting generators anyway; all emissions go the the co2 atmosphere store.

**Comment by fneum** [comment]
Great! Thanks for looking into it and sorry that the script is so buggy...

**Comment by fneum** [comment]
test failures seem unrelated, I will merge given previous commit CI ran through.

**Comment by koen-vg** [comment]
No worries, happy to help and anyway it's so much better to have anything at all to get started with than having to do everything scratch!

**Pull request merged**